core/src/main/scala/dagr/core/execsystem/TaskTracker.scala

```diff
@@ -315,4 +315,9 @@ trait TaskTracker extends TaskManagerLike with LazyLogging {
   protected def graphNodesWithPredecessors: Iterable[GraphNode] = {
     graphNodes.filter(node => GraphNodeState.hasPredecessors(node.state))
   }
+
+  protected def forallGraphNodes(states: Set[GraphNodeState.Value]): Boolean = {
```

Comment:
Since this takes a set of states rather than a predicate, how about `allGraphNodesInStates()`?